### DISCUSSION TITLE
Load symbol from mismatched symbol

### DISCUSSION THREAD

**Discussion by sgkoishi** [icsharpcode/ILSpy#2351] [Ideas]
If there are two similar assemblies of different versions and one comes with symbol, is it possible to compare the AST to get variable name hint from the symbol?
Or to load the mismatched symbol file and try the variable name at the method level.

**Comment by siegfriedpammer** [comment]
You can tell ILSpy to load a different PDB file by selecting "Select PDB..." from the assembly tree node context menu. Would that solve your problem?

**Reply by sgkoishi** [reply]
Thanks for answering! I thought it only load matching symbols and never have tried it yet.

**Reply by sgkoishi** [reply]
Found the `IDebugInfoProvider` and realize that all the local's names are located by the method and numerical index and some string got named num24 or similar - gonna try to save some kind of “detailed” version.
